cumulus/client/cirrus-service/src/lib.rs

```diff
@@ -30,48 +30,30 @@ use sc_consensus::{
 	BlockImport,
 };
 use sc_network::NetworkService;
-use sc_service::{Configuration, Role, TaskManager};
+use sc_service::{Configuration, TaskManager};
 use sc_transaction_pool_api::TransactionPool;
 use sc_utils::mpsc::tracing_unbounded;
 use sp_api::ProvideRuntimeApi;
 use sp_blockchain::HeaderBackend;
 use sp_consensus::BlockOrigin;
-use sp_core::{traits::SpawnNamed, Pair};
+use sp_core::traits::SpawnNamed;
 use sp_inherents::CreateInherentDataProviders;
 use sp_runtime::{
 	traits::{Block as BlockT, NumberFor},
 	Justifications,
 };
-use std::{ops::Deref, sync::Arc};
+use std::sync::Arc;
 
 use cumulus_client_consensus_common::RelaychainClient;
 
 pub mod genesis;
 
-use cirrus_node_primitives::CollatorPair;
-
-/// The primary chain full node handle.
-pub struct PrimaryFullNode<C> {
-	/// The relay chain full node handles.
-	pub primary_chain_full_node: subspace_service::NewFull<C>,
-	/// The collator key used by the node.
-	pub collator_key: CollatorPair,
```

Comment:
I'm not sure if the `collator_key` will be necessary eventually.

Reply:
We can add it later when we know where and how we need it. Right now it just causes us to maintain a bunch of code that would otherwise be unnecessary with extra dependencies. With all of these things interleaved it is hard to see the code logic as it is scattered across many functions and sometimes even crates.